public/app/plugins/datasource/elasticsearch/datasource.test.ts

```diff
@@ -50,13 +50,33 @@ describe('ElasticDatasource', function(this: any) {
     }),
   };
 
+  const threeDayTimeSrv: any = {
```

Comment:
Maybe just create a factory for the timeSrv that would take the `from` and `to` as `timeSrv` and `threeDayTimeSrv` are basically the same.

Reply:
Yeah, repeating the same function was definitely not a good thing, but the usage of `this` has bothered me. Would something like the below work?

```
  function createTimeSrv(from: string) {
    const srv: any = {
      time: { from: from, to: 'now' },
    };

    srv.timeRange = jest.fn(() => {
      return {
        from: dateMath.parse(srv.time.from, false),
        to: dateMath.parse(srv.time.to, true),
      };
    });

    srv.setTime = jest.fn(time => {
      srv.time = time;
    });
  }
```

I even omitted the `timeRange` and `setTime` functions and the tests seems to have passed, so I wonder if they are necessary.

Reply:
Yes probably a good call, it would not be a problem if we did not use arrow functions there.